webapp/client/src/components/NewsletterRegisterBox.js

```diff
@@ -0,0 +1,259 @@
+      var(--spacing-07b);
+  }
+`;
+
+const BoxHeadline = styled.h2`
+  margin: 0;
+  font-size: var(--text-medium-big);
+  padding-bottom: var(--spacing-03);
+
+  @media (min-width: 577px) {
+    font-size: var(--text-xxl);
+  }
+
+  @media (min-width: 769px) {
+    padding-bottom: var(--spacing-04a);
+  }
+`;
+const BoxText = styled.span`
+  font-size: var(--text-medium);
+`;
+
+const SmallText = styled.span`
+  font-size: var(--text-medium);
+  color: var(--secondary-text-color);
+`;
+
+const ButtonInRow = styled(ButtonAnchor)`
+  margin-top: var(--spacing-05);
+  @media (min-width: 577px) {
+    margin-top: 47px;
+  }
+  width: 195px;
+`;
+
+const SuccessBox = styled.div`
+  background-color: var(--white);
+  display: none;
+  margin-bottom: var(--spacing-03);
+`;
+
+const SuccessIconSmall = styled(SuccessIcon)`
+  height: 30px;
+  width: 30px;
+  top: 16px;
+  left: 24px;
+  @media (min-width: 577px) {
+    margin-left: var(--spacing-05);
+  }
+  margin-left: var(--spacing-03);
+  margin-right: var(--spacing-03);
+`;
+
+const RowFieldButton = styled.div`
+  display: grid;
+  @media (min-width: 577px) {
+    display: flex;
+  }
+  flex-wrap: wrap;
+  padding-bottom: var(--spacing-03);
+`;
+
+const ColumnField = styled.div`
+  display: inline-block;
+
+  @media (min-width: 577px) {
+    flex-basis: ${({ flexBasis }) => `${flexBasis}%`};
+  }
+
+  .text-input-field-label {
+    margin-top: 0;
+  }
+
+  .field-margin-top {
+    margin-top: var(--spacing-05) !important;
+  }
+`;
+
+const ColumnButton = styled.div`
+  display: inline-block;
+  @media (min-width: 577px) {
+    flex-basis: ${({ flexBasis }) => `${flexBasis}%`};
+    ${({ leftAuto }) =>
+      leftAuto &&
+      `
+    margin-left: auto;
+  `}
+  }
+`;
+
+const RowTickSuccessMessage = styled.div`
+  padding-bottom: var(--spacing-03);
+  display: grid;
+  @media (min-width: 577px) {
+    display: flex;
+  }
+`;
+
+const ColumnTick = styled.div`
+  display: inline-block;
+  flex-basis: ${({ flexBasis }) => `${flexBasis}%`};
+  margin-top: var(--spacing-03);
+`;
+
+const ColumnSuccessMessage = styled.div`
+  display: inline-block;
+  flex-basis: ${({ flexBasis }) => `${flexBasis}%`};
+  margin-top: var(--spacing-03);
+  @media (max-width: 576px) {
+    margin-left: var(--spacing-03);
+    margin-right: var(--spacing-03);
+  }
+`;
+
+const errors = [];
+let emailValue = "";
+
+export default function NewsletterRegisterBox({ dataPrivacyLink, csrfToken }) {
+  const { t } = useTranslation();
+
+  function displaySuccessBox(activate) {
+    const x = document.getElementById("success");
+    if (activate) {
+      x.style.display = "block";
+    } else {
+      x.style.display = "none";
+    }
+  }
+
+  const [, updateState] = React.useState();
+  const forceUpdate = React.useCallback(() => {
+    displaySuccessBox(false);
+    updateState({});
+  }, []);
+
+  const sendEmail = () => {
+    emailValue = document.getElementsByTagName("input")[0].value;
```

Comment:
you should use getElementsById instead of a tag name to be more specific.
I think if you use <form> (react: StepForm) then you don't need to handle the fetch manually here in the JS. I guess the validation and handling should be done in the backend or do I miss something?

Reply:
I will change to getElementsById (so true) but lets have a talk about doing the fetching in JS later